api/envoy/api/v2/cluster/outlier_detection.proto

```diff
@@ -76,4 +76,24 @@ message OutlierDetection {
   // used to disable ejection or to ramp it up slowly. Defaults to 0.
   google.protobuf.UInt32Value enforcing_consecutive_gateway_failure = 11
       [(validate.rules).uint32.lte = 100];
+
+  // The number of consecutive connect failures before ejection
+  // occurs. Defaults to 5.
+  google.protobuf.UInt32Value consecutive_connect_failure = 12;
+
+  // The % chance that a host will be actually ejected when an outlier status
+  // is detected through consecutive connect failures. This setting can be
+  // used to disable ejection or to ramp it up slowly. Defaults to 100.
+  google.protobuf.UInt32Value enforcing_consecutive_connect_failure = 13
+      [(validate.rules).uint32.lte = 100];
+
+  // The number of consecutive server transaction failures before ejection occurs.
```

Comment:
Just from reading this documentation, I'm not sure what a server transaction failure is. I will keep reading the PR but can we update that to make it more clear?

Edit: OK So the idea here is this is an upstream server failure for non-HTTP servers? Is that right? If so I think that makes sense.

Reply:
That is correct. Upstream server failure would be a situation when connection was OK, but for example redis DB transaction failed. Server could not understand what it received or parsing of reply failed.